libraries/azure_ad_user.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class AzureAdUser < AzurermResource
+  name 'azure_ad_user'
+  desc 'Verifies settings for an Azure Active Directory User'
+  example "
+    describe azure_ad_user(object_id: 'object_id') do
+    end
+  "
+
+  ATTRS = %i(
+    objectId
+    mail
+    displayName
+    userType
+  ).freeze
+
+  attr_reader(*ATTRS)
+
+  # Can be constructed either via API call with an Azure ID provided,
+  # or from parameters passed. API call takes precedence when ID provided.
+  def initialize(azure_id, user_args)
```

Comment:
In the convention we've been using, this should only ever take an ID. (See comments elsewhere about collections just containing IDs.)